cpp/tests/strings/translate_tests.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/translate.hpp>
+#include <cudf/utilities/error.hpp>
+
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include "./utilities.h"
+
+#include <vector>
+#include <gmock/gmock.h>
+
+
+struct StringsTranslateTest : public cudf::test::BaseFixture {};
+
+std::pair<cudf::char_utf8,cudf::char_utf8> make_entry( const char* from, const char* to )
+{
+    cudf::char_utf8 in, out = 0;
+    cudf::strings::detail::to_char_utf8(from, in);
+    if( to )
+        cudf::strings::detail::to_char_utf8(to, out);
+    return std::make_pair(in,out);
+}
```

Comment:
The `to_char_utf8()` and `from_char_utf8()` are more like `memcpy` with a copy size since multiple bytes may be copied between the two pointers. I'd rather keep the signatures of these two methods in sync.

Reply:
What represents the size here? I don't think we should be designing functions / code to be similar to C functions like `memcpy`

Reply:
The size is encoded in the character itself. Most of the time I need to know the size after using these two functions (`to_char_utf8` and `from_char_utf8`) and since the size is computed during their execution, they just return it. Otherwise, I would not have themreturn anything for these two functions.